libnavigation-core/src/main/java/com/mapbox/navigation/core/navigator/LocationEx.kt

```diff
@@ -48,11 +51,48 @@ internal fun Location.toFixLocation(): FixLocation {
         provider,
         bearingAccuracy,
         speedAccuracy,
-        verticalAccuracy
+        verticalAccuracy,
+        if (extras != null) extras.toMap() else Bundle().toMap()
     )
 }
 
 internal fun List<FixLocation>.toLocations(): List<Location> = this.map { it.toLocation() }
 
 private fun isCurrentSdkVersionEqualOrGreaterThan(sdkCode: Int): Boolean =
     Build.VERSION.SDK_INT >= sdkCode
+
+internal fun HashMap<String, Value>.toBundle(): Bundle? {
+    if (this.isEmpty()) return null
+    val bundle = Bundle()
+    for ((key, value) in this.entries) {
+        when (val contents = value.contents) {
+            is Double -> bundle.putDouble(key, contents)
+            is Long -> bundle.putLong(key, contents)
+            is Boolean -> bundle.putBoolean(key, contents)
+            is String -> bundle.putString(key, contents)
+            else -> {
+                // do nothing
+            }
+        }
+    }
+    return bundle
+}
+
+internal fun Bundle.toMap(): HashMap<String, Value> {
+    val map: HashMap<String, Value> = HashMap()
+    val keySet = this.keySet()
+    val iterator: Iterator<String> = keySet.iterator()
+    while (iterator.hasNext()) {
+        val key = iterator.next()
+        when (val value = this.get(key)) {
+            is Double -> map[key] = Value(value)
+            is Long -> map[key] = Value(value)
+            is Boolean -> map[key] = Value(value)
+            is String -> map[key] = Value(value)
+            else -> {
+                // do nothing
+            }
```

Comment:
Same as in https://github.com/mapbox/mapbox-navigation-android/pull/5034/files#r747684218